compiler/onecc-docker/onecc-docker

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import subprocess
+import json
+import requests
+
+
+def _run(cmd, is_shell=False):
+    result = subprocess.Popen(
+        cmd, shell=is_shell, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    stdout, stderr = result.communicate()
+    stdout = stdout.decode('utf-8')
+    stderr = stderr.decode('utf-8')
+
+    if result.returncode:
+        print(stderr, end='')
+        exit(result.returncode)
+    else:
+        return stdout
+
+
+def _image_exists(name):
+    cmd = ['docker', 'images', '-q', name]
+    lines = _run(cmd).splitlines()
+    return lines
+
+
+def main():
+    # TODO Update Dockerfile path
+    dockerfile_path = "."
+
+    TAG_URL = "https://api.github.com/repos/Samsung/ONE/releases/latest"
+    try:
+        response = requests.get(TAG_URL)
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        SystemExit('onecc-docker: error: {}'.format(e))
+
+    versions_str = response.content
+    versions_json = json.loads(versions_str)
+    recent_version = versions_json["tag_name"]
+
+    image_name = f"onecc:{recent_version}"
+    build_arg = f"VERSION={recent_version}"
+
+    if not _image_exists(image_name):
+        build_cmd = [
+            "docker", "build", "-t", image_name, "--build-arg", build_arg, dockerfile_path
+        ]
+        print('build Docker image ...')
+        _run(build_cmd)
+        print('Dockerfile successfully built.')
+
+    contianer_name = f"onecc_{recent_version.replace('.','_')}"
+    user_cmd = ''.join(sys.argv[1:])
+
+    run_cmd = [
+        "docker", "run", "--rm", "-u", "$(id -u):$(id -g)", "--name", contianer_name,
+        "-v", "${HOME}:${HOME}", "-e", "HOME=${HOME}", "-w", "${PWD}", image_name,
+        user_cmd
+    ]
+
+    cmd = ' '.join(run_cmd)
+    output = _run(cmd, is_shell=True)
```

Comment:
What is this `is_shell` for?

Reply:
Good question.
And will there be `is_shell=True` for future new codes?

Reply:
Options for calling with shell commands.
I tried to use the subprocess of the function `_run` as if I were entering it in the shell, depending on the situation.

Reply:
> depending on the situation.

Actually, I just wanted to know which situation are you going to handle with this `is_shell`. I guess it is for evaluating the shell variables like `${HOME}`, `${PWD}`.

Reply:
We can implement this code without using `shell=True`.

```
import docker

    home=os.environ['HOME']
    pwd=os.environ['PWD']
    client = docker.from_env()
    container = client.containers.run(image_name, user_cmd, auto_remove=True, detach=True,
        user=os.getuid(), group_add=[os.getgid()],
        volumes=['{}:{}'.format(home, home)],
        environment=["HOME={}".format(home)], working_dir=pwd)

    for line in container.logs(stream=True):
        print(line.decode("utf-8").strip())
```

But this code adds new dependencies `docker` python package.
So if possible, I want to be reflected in the existing code. If we need to modify it later, I will update it.